dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/NormalizeLoops.java

```diff
@@ -69,5 +104,46 @@ public void run(Function function) {
 
             counter++;
         }
+
+        // Guarantees header is the only entry point
+        for (Label label : function.getEvents(Label.class)) {
```

Comment:
It might be worthwhile to refactor the two parts of the pass into two methods: `makeUniqueBackjump` and `makeUniqueEntryPoint` (or similar).

Reply:
I actually ended up splitting the logic into two passes: one to guarantee single entry and another to guarantee single unconditional backjump